src/isar/apis/schedule/start_mission.py

```diff
@@ -41,14 +44,23 @@ def post(
         except MissionPlannerError as e:
             message = StartMissionMessages.mission_not_found()
             self.logger.error(e)
-            return JSONResponse(
-                content=asdict(message), status_code=HTTPStatus.NOT_FOUND
-            )
+            response.status_code = HTTPStatus.NOT_FOUND.value
+            return StartResponse(message=message.message, started=message.started)
 
-        ready, response = self.scheduling_utilities.ready_to_start_mission()
+        ready, response_scheduler = self.scheduling_utilities.ready_to_start_mission()
         if ready:
-            response = self.scheduling_utilities.start_mission(mission=mission)
-            self.logger.info(response)
+            response_scheduler = self.scheduling_utilities.start_mission(
+                mission=mission
+            )
+            self.logger.info(response_scheduler)
 
-        message, status_code = response
-        return JSONResponse(content=asdict(message), status_code=status_code)
+        message, status_code = response_scheduler
+        response.status_code = status_code.value
+        if status_code == HTTPStatus.OK:
+            return StartResponse(
+                message=message.message,
+                started=message.started,
+                mission_id=str(mission.id),
+            )
+        else:
+            return StartResponse(message=message.message, started=message.started)
```

Comment:
Have another look at this section. I am a bit confused by some parts of it, see if it can be made a bit clearer.